pcap/stream.go

```diff
@@ -219,12 +227,18 @@ func (f *tcpFlow) reassemblyComplete() {
 }
 
 func (f *tcpFlow) toPNT(firstPacketTime time.Time, lastPacketTime time.Time,
-	c akinet.ParsedNetworkContent) akinet.ParsedNetworkTraffic {
-	if firstPacketTime.IsZero() {
+	c akinet.ParsedNetworkContent,
+) akinet.ParsedNetworkTraffic {
+	if firstPacketTime.IsZero() || lastPacketTime.IsZero() {
 		firstPacketTime = f.clock.Now()
+		lastPacketTime = firstPacketTime
```

Comment:
Oops, taking another look at this -- you check whether either is zero but this might set both to zero.

We should at least print out in the log what the values were _before_ making them equal.  But if we have one zero and one nonzero, what should we do?

If firstPacketTime is zero, then it feels like we don't end up with a valid ClientWitnessTime at the end, is there a check later covering this?  I guess I'm unclear whether this matches the intention you had described yesterday.

Reply:
Logged the timestamps before modification and I made sure to only modify the ones that are zero value.